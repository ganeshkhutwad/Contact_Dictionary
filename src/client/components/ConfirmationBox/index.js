export { default as ConfirmationBox } from './ConfirmationBox';