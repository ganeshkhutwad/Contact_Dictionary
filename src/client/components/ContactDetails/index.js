export { default as ContactDetails } from './ContactDetails';