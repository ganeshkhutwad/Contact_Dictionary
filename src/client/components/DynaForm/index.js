export { default as DynaForm } from './DynaForm';
