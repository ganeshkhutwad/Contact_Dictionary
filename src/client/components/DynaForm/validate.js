/**
@author Ganesh Khutwad
 */
const validate = {
    required: value => (value || typeof value === 'number' ? undefined : 'Required'),

    maxLength: max => value => value && value.length > max ? `Must be ${max} characters or less` : undefined,

    get maxLength15() {
        return this.maxLength(15);
    },

    minLength: min => value => value && value.length < min ? `Must be ${min} characters or more` : undefined,

    get minLength2() {
        return this.minLength(2);
    },

    number: value => value && isNaN(Number(value)) ? 'Must be a number' : undefined,

    minValue: min => value => value && value < min ? `Must be at least ${min}` : undefined,

    get minValue13() {
        return this.minValue(13);
    },

    email: value => value && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value) ? 'Invalid email address' : undefined,

    alphaNumeric: value => value && /[^a-zA-Z0-9 ]/i.test(value) ? 'Only alphanumeric characters' : undefined,

    phoneNumber: value => value && !/^(0|[1-9][0-9]{9})$/i.test(value) ? 'Invalid phone number, must be 10 digits' : undefined
}

export default validate;
