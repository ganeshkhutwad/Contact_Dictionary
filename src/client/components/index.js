export { Header } from './Header';
export { ContactList } from './ContactList';
export { ContactDetails } from './ContactDetails';
export { AddButton } from './AddButton';
