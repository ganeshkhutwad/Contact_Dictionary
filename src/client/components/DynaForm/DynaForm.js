/**
@author Ganesh Khutwad
 */
import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import TextField from 'material-ui/TextField';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import Button from '@material-ui/core/Button';
import validate from './validate';

const btn = {
    margin: '10px'
};
const fieldWidth = {
    width: '100%'
};

const renderTextField = (
  { input, label, meta: { touched, error }, ...custom },
) => (
  <TextField
    hintText={label}
    floatingLabelText={label}
    errorText={touched && error}
    {...input}
    {...custom}
  />
);

const renderSelectField = (
  { input, label, meta: { touched, error }, children, ...custom },
) => (
  <SelectField
    floatingLabelText={label}
    errorText={touched && error}
    {...input}
    onChange={(event, index, value) => input.onChange(value)}
    children={children}
    {...custom}
  />
);

const { required, maxLength15, minLength2, alphaNumeric, email, number, phoneNumber } = validate;
class DynaForm extends Component {

    componentDidMount() {
        this.handleInitialize();
    }

    handleInitialize = () => {
        this.props.initialize(this.props.formValues);
    }

  render() {
      const { handleSubmit, pristine, reset, submitting } = this.props;
      return (
        <form onSubmit={handleSubmit} autocomplete="off">
        <div>
            <Field
            style={fieldWidth}
            name="id"
            component={renderTextField}
            label="Id"
            disabled="true"
            />
        </div>
        <div>
            <Field
            style={fieldWidth}
            name="firstName"
            component={renderTextField}
            label="First Name"
            validate={[required, maxLength15, minLength2, alphaNumeric]}
            />
        </div>
        <div>
            <Field style={fieldWidth} name="lastName" component={renderTextField} label="Last Name" validate={[required, maxLength15, minLength2, alphaNumeric]} />
        </div>
        <div>
            <Field style={fieldWidth} name="email" component={renderTextField} label="Email" validate={[required, email]}/>
        </div>
        <div>
            <Field validate={[required, phoneNumber]} style={fieldWidth} name="mobileNum" component={renderTextField} label="Mobile Number" />
        </div>
        <div>
            <Field
            style={fieldWidth}
            name="status"
            component={renderSelectField}
            label="Status"
            validate={required}
            >
            <MenuItem value="active" primaryText="Active" />
            <MenuItem value="inactive" primaryText="Inactive" />
            </Field>
        </div>
        <div>
            <Button type="submit" disabled={pristine || submitting} variant="contained" color="primary" style={btn}>
                Submit
            </Button>
            <Button onClick={reset} disabled={pristine || submitting} variant="contained" style={btn}>
                Reset
            </Button>
        </div>
        </form>
    );
  }
};

export default reduxForm({
  form: 'DynaForm', // a unique identifier for this form
})(DynaForm);
