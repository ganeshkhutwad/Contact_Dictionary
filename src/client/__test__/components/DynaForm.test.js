/**
Test `AddButton` Presentational Component.
@author Ganesh Khutwad
 */
import React from 'react';
import { DynaForm } from '../../components/DynaForm';

const classes = { };

describe('`DynaForm` Tests', () => {
    it('`DynaForm` Snapshot', () => {
        const dynaFormWrapper = shallow(<DynaForm classes={classes} />);
        expect(dynaFormWrapper).toMatchSnapshot();
    });
});
