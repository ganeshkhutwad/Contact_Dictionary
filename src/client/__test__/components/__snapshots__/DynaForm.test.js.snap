// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`DynaForm\` Tests \`DynaForm\` Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm
    classes={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {},
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "DynaForm",
      "getFormState": [Function],
      "initialValues": undefined,
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {},
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "DynaForm",
        "getFormState": [Function],
        "initialValues": undefined,
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
